app/models/results/response_json_generator.rb

```diff
@@ -49,7 +50,37 @@ def add_group_answers(group, object)
         object << (item = {})
         add_answers(group, item)
       else
-        add_answers(group, object[group_key(group)] = {})
+        subgroup = object[node_key(group)] = {}
+        add_answers(group, subgroup)
+      end
+    end
+
+    # Make sure we include everything specified by metadata in our output,
+    # even if an older Response didn't include that qing/group originally.
+    def add_nil_answers(node, object)
```

Comment:
Can we call this var `item` or `form_item` to distinguish it from `node` which is short for `ResponseNode`.

Also, looking back, `object` is a terrible var name to use throughout here. My bad. `json_object` would be much preferable. Would you mind fixing?